cdap-ui/app/cdap/components/EntityListView/index.js

```diff
@@ -135,23 +135,22 @@ class EntityListView extends Component {
      );
     }
 
-    let query = nextProps.location.query || {};
-    let {filter = defaultFilter, q , page = 1, sort, order } = query;
-    let sortObj;
-    if (typeof q === 'undefined'){
-      let matchedSort = this.sortOptions.find(sortOption => sortOption.sort === sort && sortOption.order === order);
-      sortObj = matchedSort ? matchedSort : this.sortOptions[0];
-    } else {
-      sortObj = this.sortOptions[0];
+    let queryObject = this.getQueryObject(nextProps.location.query);
+    if (
+      queryObject.filter !== this.state.filter &&
+      queryObject.sort.fullSort !== this.state.sortObj.fullSort &&
+      queryObject.query !== this.state.query &&
+      queryObject.page !== this.state.currentPage
+    ) {
+      this.updateData(queryObject.query, queryObject.filter, queryObject.sort, nextProps.params.namespace, queryObject.page);
```

Comment:
should we reverse the order of `updateData` and `setState`, so tin case this code gets called twice in a row (multiple clicks), it wont call updateData multiple times?